docs/source/manual/configuration.rst

```diff
@@ -767,6 +767,113 @@ Name                   Default      Description
 type                   REQUIRED     The filter type.
 ====================== ===========  ================
 
+.. _man-configuration-json-layout:
+
+JSON layout
+------------
+
+.. code-block:: yaml
+
+    layout:
+      type: json
+      timestampFormat: "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+      includeTimestamp: true
+      prettyPrint: false
+      appendLineSeparator: true
+      includeLevel: true
+      includeThreadName: true
+      includeMDC: true
+      includeLoggerName: true
+      includeMessage: true
+      includeException: true
+      includeContextName: false
+      customFieldNames:
+        timestamp: "@timestamp"
+      additionalFields:
+        service-name: "user-service"
+
+=======================  ===========  ================
+Name                     Default      Description
+=======================  ===========  ================
+includeTimestamp         true         Whether to include the timestamp of the event in the JSON map.
+timestampFormat          (none)       By default, the timestamp is not formatted; To format the timestamp using set the property with the
+                                      corresponding ``DateTimeFormatter`` string, for example, {@code yyyy-MM-dd HH:mm:ss.SSSZ}.
+prettyPrint              false        Whether the JSON output should be formatted for human readability.
+appendLineSeparator      true         Whether to append a line separator at the end of the message formatted as JSON.
+customFieldNames         (empty)      A map of field name replacements in the generated map. For example ``requestTime:request_time, userAgent:user_agent)``
+includeLevel             true         Whether to include the logging level in the JSON map as the ``level`` field.
+includeThreadName        true         Whether to include the thread name in the JSON map as the ``thread`` field.
+includeMdc               true         Whether to include the MDC properties in the JSON map as the ``mdc`` field.
+includeLoggerName        true         Whether to include the logger name in the JSON map as the ``logger`` field.
+includeMessage           true         Whether to include the formatted message in the JSON map as the ``message`` field ..
+includeException         true         Whether to log exceptions. If the property enabled and there is an exception,
+                                      it will be formatted to a string added in the JSON map as the ``exception`` field.
+customFieldNames         (empty)      A map of field name replacements in the JSON map. For example ``requestTime:request_time, userAgent:user_agent)``.
+additionalFields         (empty)      A map of fields to add in the JSON map.
+
+.. _man-configuration-json-access-layout:
+
+JSON access log layout
+------------
+
+.. code-block:: yaml
+
+    layout:
+      type: access-json
+      timestampFormat: "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+      prettyPrint: false
+      includeTimestamp: true
+      appendLineSeparator: true
+      includeRemoteAddr: false
```

Comment:
The include prefixes seem unnecessary.

Reply:
It looks to me it makes sense to represent included boolean options as a set:
```yaml
include:
  - remoteAddress
  - remoteUser
  - requestTime
```
or
```yaml
include: [remoteAddress, remoteUser, requestTime]
```